PANIC when a job is closing

When job master meets error and closes itself, it will PANIC.
Additionally, the panic log WILL NOT be printed in the files. In stderr/stdio file, it shows the real panic log.

log is 
```
[2022/03/08 00:14:08.344 +08:00] [INFO] [server.go:535] ["peer connection received"] [senderID=48b3bfcf-5ca0-4484-ab5f-7c16d61f9c49] [senderAdvertiseAddr=10.244.5.88:10241] [addr=10.244.5.88:56380] [epoch=1]
[2022/03/08 00:14:08.386 +08:00] [INFO] [server.go:535] ["peer connection received"] [senderID=11195c75-92e5-438d-a08f-315f17c4f319] [senderAdvertiseAddr=10.244.4.74:10241] [addr=10.244.4.74:53158] [epoch=1]
[2022/03/08 00:14:09.430 +08:00] [INFO] [server.go:535] ["peer connection received"] [senderID=7ccdcf7f-e4f1-46ce-b05b-51c31a3a8e3e] [senderAdvertiseAddr=10.244.6.55:10241] [addr=10.244.6.55:38218] [epoch=1]
[2022/03/08 00:14:10.335 +08:00] [WARN] [server.go:438] ["topic handler returned error"] [error="[CDC:ErrWorkerPoolHandleCancelled]workerpool handle is cancelled"] [errorVerbose="[CDC:ErrWorkerPoolHandleCancelled]workerpool handle is cancelled\ngithub.com/pingcap/errors.AddStack\n\t/root/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\t/root/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/workerpool.(*defaultEventHandle).GracefulUnregister.func1\n\t/root/go/pkg/mod/github.com/pingcap/tiflow@v0.0.0-20220224093943-901ba1d0ff05/pkg/workerpool/pool_impl.go:201\ngithub.com/pingcap/tiflow/pkg/workerpool.(*defaultEventHandle).GracefulUnregister\n\t/root/go/pkg/mod/github.com/pingcap/tiflow@v0.0.0-20220224093943-901ba1d0ff05/pkg/workerpool/pool_impl.go:223\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageServer).run.func2\n\t/root/go/pkg/mod/github.com/pingcap/tiflow@v0.0.0-20220224093943-901ba1d0ff05/pkg/p2p/server.go:240\nruntime.goexit\n\t/usr/lib/golang/src/runtime/asm_amd64.s:1371"]
[2022/03/08 00:14:11.974 +08:00] [INFO] [server.go:535] ["peer connection received"] [senderID=ip1-40497c16-17c9-476a-abb5-c393ebfcebc5] [senderAdvertiseAddr=10.244.6.55:10240] [addr=10.244.6.55:38594] [epoch=1]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3d87adb]

goroutine 248 [running]:
github.com/hanfei1991/microcosm/lib.(*DefaultBaseWorker).doClose(0xc0002b31d0)
        /root/microcosm/lib/worker.go:237 +0x7b
github.com/hanfei1991/microcosm/lib.(*DefaultBaseJobMaster).Close(0xc000b97d80, 0x52d5a48, 0xc000f6ad00, 0x0, 0x2)
        /root/microcosm/lib/base_jobmaster.go:157 +0xad
github.com/hanfei1991/microcosm/executor/worker.(*Runtime).closeTask(0xc0004e3200, 0x52d5a48, 0xc000f6ad00)
        /root/microcosm/executor/worker/runtime.go:143 +0x118
github.com/hanfei1991/microcosm/executor/worker.(*Runtime).Start.func1(0xc000bf5370, 0xc0004e3200, 0x52d5a48, 0xc000f6ad00)
        /root/microcosm/executor/worker/runtime.go:188 +0x6b
created by github.com/hanfei1991/microcosm/executor/worker.(*Runtime).Start
        /root/microcosm/executor/worker/runtime.go:186 +0x8d
~
```

--- a/lib/worker.go
+++ b/lib/worker.go
@@ -234,8 +234,12 @@ func (w *DefaultBaseWorker) Poll(ctx context.Context) error {
 
 func (w *DefaultBaseWorker) doClose() {
 	w.cancelMu.Lock()
-	w.cancelBgTasks()
-	w.cancelPool()
+	if w.cancelBgTasks != nil {
+		w.cancelBgTasks()
+	}
+	if w.cancelPool != nil {
+		w.cancelPool()
+	}
 	w.cancelMu.Unlock()
 
 	closeCtx, cancel := context.WithTimeout(context.Background(), time.Second*3)

--- a/lib/worker_test.go
+++ b/lib/worker_test.go
@@ -315,3 +315,14 @@ func TestWorkerSuicide(t *testing.T) {
 
 	require.Regexp(t, ".*Suicide.*", exitErr.Error())
 }
+
+func TestCloseBeforeInit(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	worker := newMockWorkerImpl(workerID1, masterName)
+
+	worker.On("CloseImpl").Return(nil)
+	err := worker.Close(ctx)
+	require.NoError(t, err)
+}

